src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,43 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive(animals) {
+    const alive = animals.filter(animal => animal.health > 0);
+
+    return alive;
+  }
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(bited) {
+    if (!bited.hidden && bited.hasOwnProperty('hidden')) {
+      bited.health -= 50;
+    }
+
+    if (bited.health === 0) {
```

Comment:
sometimes in other scenarios it will not work
so fo sure it is better check in this way
```suggestion
    if (animal.health <= 0) {
```